src/sentry/api/endpoints/organization_missing_org_members.py

```diff
@@ -56,12 +59,31 @@ def _get_missing_members(self, organization: Organization) -> QuerySet[CommitAut
             .order_by("-commit_count")
         )
 
-    # TODO(cathy): check domain
+    def _get_email_domain(self, email: str) -> str:
+        return Address(addr_spec=email).domain
 
     def get(self, request: Request, organization) -> Response:
-        # TODO(cathy): search
         queryset = self._get_missing_members(organization)
 
+        # if a member has user_email=None, then they have yet to accept an invite
+        org_owners = organization.get_members_with_org_roles(
+            roles=[roles.get_top_dog().id]
+        ).exclude(user_email=None)
+
+        owner_emails = {self._get_email_domain(owner.user_email) for owner in org_owners}
+
+        if len(owner_emails) == 1:
+            queryset = queryset.filter(email__endswith=owner_emails.pop())
```

Comment:
If you move all business logic into a separate function, it will be easier to grok this GET endpoint.

```suggestion
def get(self, request, ...):
    queryset = self._get_missing_members(organization)
    
    common_domain = self._get_common_email_domain(...)
    if common_domain:
        queryset = queryset.filter(email__endswith=common_domain)
        
    query = ...
    if query:
        ...
    
    return Response(...)
    
    
def _get_common_email_domain(self, ...):
    """
    If all the owners in an org has the same email domain, we'll return that value and assume that everyone in the org should be using that email domain
    """
    # if a member has user_email=None, then they have yet to accept an invite
    org_owners = organization.get_members_with_org_roles(
        roles=[roles.get_top_dog().id]
    ).exclude(user_email=None)
    
    def _get_email_domain():
        ....

    owner_emails = {_get_email_domain(owner.user_email) for owner in org_owners}

    if len(owner_emails) == 1:
        return owner_emails.pop()
    return None
```

Reply:
Also, should this filtering always happen? Is there a chance that I'll want to search for a user that doesn't have the same domain?

Reply:
hmm the filtering is in place to protect against bots or users that aren't actually affiliated with the organization. i'm not actually sure which Github email gets stored in the `CommitAuthor` table when somebody makes a commit, though